### ISSUE TITLE
Here are some notes for a better port

### ISSUE THREAD

**Issue by Coderx7** [mrecos/SimpNet_Keras_R#1]
First of all thank you for your effort on porting SimpNet. 
I had a quick look at your scripts, and noticed there were several pieces missing. 
For CIFAR10/100, you need to preprocess the dataset. normalize and zero-pad them. You can have a look at this script [here](https://github.com/Pastromhaug/caffe-stochastic-depth/blob/master/examples/cifar10/preprocessing.py)  . 
Then you can use horizontal flipping and start your training (with random crops of 32x32 as is done in Caffe) 
You also need to pay attention to the weight decay and the way the optimization is carried out. As for the 8.9M experiment, I guess I used these parameters: 

```
base_lr: 0.1
lr_policy: "multistep"
gamma: 0.1
momentum: 0.9
weight_decay: 0.005
delta: 0.001
stepvalue: 50000
stepvalue: 95000
stepvalue: 153000
stepvalue: 195000
stepvalue: 220000
stepvalue: 270000
type: "AdaDelta"
```
You can also lower the weight decay to sth like 0.0005 but increase the dropout ratio e.g. for the layers that have more neurons (i.e the last 3,4 layers), (this will be more effective when you have a warm start). 
I'm dont know Keras, so in porting different hyper parameters from Caffe to Keras, I cant really help, anything else, please feel free to contact me (preferably through email : Coderx7 at (Gmail.com) 

Good luck

**Comment by mrecos** [comment]
@Coderx7 Thank you very much for the information! I have implemented most of the augmentation; although Keras does not naively do random crop at this time. I saw the weight decay in your code, but have not yet figured that out in this setting. I will continue to work on these and hopefully end up at a faithful port. Thanks!
